api/places365/places365.py

```diff
@@ -117,22 +122,34 @@ def load_model():
         res = {}
 
         # output the IO prediction
-        io_image = np.mean(labels_IO[idx[:10]]) # vote for the indoor or outdoor
+        # labels_IO[idx[:10]] returns a list of 0's and 1's: 0 -> inside, 1 -> outside
+        # Determine the mean to reach a consensus
+        io_image = np.mean(labels_IO[idx[:10]])
         if io_image < 0.5:
             res['environment'] = 'indoor'
         else:
             res['environment'] = 'outdoor'
 
         # output the prediction of scene category
+        # idx[i] returns a index number for which class it corresponds to
+        # classes[idx[i]], thus returns the class name
+        # idx is sorted together with probs, with highest probabilities first
         res['categories'] = []
         for i in range(0, 5):
             if probs[i] > confidence:
                 res['categories'].append(remove_nonspace_separators(classes[idx[i]]))
+            else:
+                break
 
         # output the scene attributes
+        # This is something I don't quiet grasp yet
+        # Probs is not usable here anymore, we're not processing our input_image
+        # Take the dot product of out W_attribute model and the feature blobs
+        # And sort it along the -1 axis
+        # This results in idx_a, with the last elements the index numbers of attributes, we have the most confidence in
+        # Can't seem to get any confidence values, also all the attributes it detect are not really meaningful i.m.o.
```

Comment:
I think the original author copy and pasted part of this file here: https://github.com/CSAILVision/places365/blob/master/demo_pytorch_CAM.py
It seems to output a .json file. Maybe it contains some probability value.
 
While it is nice to have general attributes, it isn't as nice as object detection.